pipeline/wj_luigi.py

```diff
@@ -0,0 +1,167 @@
+        return LoadToDB(csv_file="data/trip.csv", table_name="trip"), LoadToDB(csv_file="data/zone.csv", table_name="zone")
+
+    def output(self):
+        return luigi.LocalTarget("./data/output.csv")
+
+    def run(self):
+        print('\n\n----------------------------Query To Database--------------------------------\n') 
+        print('\n\n--------------Upper East Side South -> Upper East Side North-----------------\n') 
+
+        query = '''
+                SELECT tpep_pickup_datetime, tpep_dropoff_datetime, passenger_count, total_amount, trip_distance, TIMESTAMPDIFF(MINUTE, tpep_pickup_datetime, tpep_dropoff_datetime) AS trip_duration
+                FROM (
+                        SELECT tpep_pickup_datetime, tpep_dropoff_datetime, passenger_count, total_amount, trip_distance, a.Zone AS PULocationZone, b.Zone AS DOLocationZone
+                        FROM `trip`
+                        JOIN `zone` a ON trip.PULocationID = a.LocationID
+                        JOIN `zone` b ON trip.DOLocationID = b.LocationID
+                    ) a
+                WHERE PULocationZone = "Upper East Side South" AND DOLocationZone = "Upper East Side North";
+                '''
+
+        output_df = pd.read_sql(query, con=mydb().engine)
+        print(output_df)
+        print('\n\n -------------------- Query Data Done----------------------------')
+
+        with self.output().open('w') as f:
+            f.write(output_df.to_csv(index=None))
+
+        return output_df
+
+
+class EmailResult(luigi.Task):
+    task_namespace = "wj_luigi"
+    csv_file = luigi.Parameter()
+
+    def output(self):
+        return luigi.LocalTarget(f"/home/weijie/luigi/simple-luigi/data/email_result_{datetime.now().strftime('%Y%m%d')}.txt")
+
+    def run(self):
+        # Create EmailMessage object
+        print('----------------------Send Email---------------------------')        
+
+        df = pd.read_csv(self.csv_file, index_col=0)
+
+        # Email content   
+        newline = '\n'
+        receipients = ['weijie@postpay.asia']
+        subject = 'Exercise'
+        message = f'''
+        Dear All,
+
+        I found something,
+
+        From "Upper East Side South" -> "Upper East Side North": \n
+        Number of rows selected: {df.shape[0]} \n
+        Average passenger count per trip: {statistics.mean(df['passenger_count'])} \n
+        Average total amount per trip: {statistics.mean(df['total_amount'])} \n
+        Average distance per trip: {statistics.mean(df['trip_distance'])} \n
+        Average duration per trip (in minutes): {statistics.mean(df['trip_duration'])} \n
+
+        -------------------------------------------
+        Summary (First 5 rows order by pickup time)
+        -------------------------------------------
+        {'|'.join(df.columns.to_list())}
+        {newline.join(
+            list("|".join(map(
+                str, df.values.tolist()[:5][i])) 
+                for i in range(5)))}
+        '''
```

Comment:
![image](https://user-images.githubusercontent.com/27208744/175917514-acb945a4-62e7-45ea-9939-a5de3bd8b8d7.png)
Please check your indentation, your email look like this.

Reply:
Email message indentation issues fixed.
![image](https://user-images.githubusercontent.com/74594391/176073599-9cf4c63a-d08b-4e70-80e8-4526e87263c7.png)